packages/k-ramel/API.md

```diff
@@ -17,3 +17,83 @@ All options are optionals to keep the default usage as simple as possible.
+ - `types.object()`: a simple object, default is `{}`
+ - `types.string()`: a simple string, default is `''`
+ - `types.bool()`: a simple boolean, default is `false`
+ - `types.array()`: a simple array, default is `[]`
+ - `types.number()`: a simple number, default is `0`
+ - `types.keyValue()`: a key value collection
+
+You can find all associated selectors and actions for each types in the [k-redux-factory documentation](https://github.com/alakarteio/k-redux-factory/blob/master/TYPES.md).
+
+### when(\<matcher1>, \<matcher2>(, ...))(\<reaction>)
+```js
+import { when } from 'k-ramel'
+```
+
+When are used to listen to redux eventbus, and then react to it.
+A `matcher` can be one of these types:
+ - a `string`: in which case the action.type will be tested against your given string
+    - ex: `when('@@krml/INIT')` will react to action `{ type: '@@krml/INIT', payload: 'whatever' }`
+ - a `regexp`: in which case the action.type will be tested against your given regexp
+    - ex: `when(/@@krml.*/)` will react to action `{ type: '@@krml/WHATEVER', payload: 'whatever' }`
+ - a `function(action, store)`: in which case the function should return a truthy value to trigger the reaction.
+
+A `reaction` must follow this signature: `(action, store, drivers) `
+
+#### Exemple 1: with a string
+In this example we react to the k-ramel initialisation to load the **Adventure time** show description from **TVMAZE**.
+
+```js
+when('@@krml/INIT')((action, store, drivers) => {
+  drivers.http('TVMAZE').get('http://api.tvmaze.com/shows/290')
+})
+```
+
+#### Exemple 2: with a regexp
+In this example we react to the http driver response to retrieve **Adventure time** show description that the API responds.
+We take care of looking for action.status so we make sure the request is ok!
+
+```js
+when(/@@http\/.*TVMAZE.*/)((action, store) => {
+  // if the request is failed, we are doing nothing
+  if (action.status < 200 || action.status >= 400) return
+
+  // in others cases we register our show description to the store
+  store.show.set(action.payload)
+})
+```
+
+#### Example 3: with a function
+We are doing the same example given previously but using a function matcher.
+ - The first matcher used is a regexp (like previous exemple): we are making sure to process a HTTP action.
+ - The second matcher is a function to make sure the request is not failed!
+    * Then, we don't need to check this in our reaction
+
+
+
+```js
+when(
+  /@@http\/.*TVMAZE.*/,
+  (action) => (action.status >= 200 && action.status < 400),
+)((action, store) => {
+  // we register our show description to the store
+  store.show.set(action.payload)
+})
+```
+
+#### Example 4: with a function, use the store
+In the function matcher you have access to the store.
+It can be useful when you want to match a specific Redux action since our actions types are published. (if you use `types`).
+
+```js
+when((action, store) => action.type === store.show.SET)(() => { console.log('show is set!') })
```

Comment:
It's cool to see that it's already possible to do it :-)

I will try to use k-ramel for a while in a project and I will make a proposition if I find it useful to add a custom matcher for a better readability.

Reply:
thank you

Reply:
If we add an helper, we can add it in a minor version